### PR TITLE
MOSIP-11739 Code changes

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java [hunk 1]
```diff
@@ -112,9 +137,57 @@ private void processCompleteAsProcessed(Exchange exchange) throws JsonProcessing
...
+		JsonObject json = new JsonObject(message);
+		WorkflowInternalActionDTO workflowEventDTO = new WorkflowInternalActionDTO();
+		workflowEventDTO.setRid(json.getString("rid"));
+		workflowEventDTO.setActionCode(WorkflowInternalActionCode.RESUME_PARENT_FLOW.toString());
+		workflowEventDTO.setActionMessage(PlatformSuccessMessages.PACKET_RESUME_PARENT_FLOW.getMessage());
+		workflowEventDTO.setReg_type(json.getString("reg_type"));
+		workflowEventDTO.setIteration(json.getInteger("iteration"));
+		workflowEventDTO.setSource(json.getString("source"));
+		exchange.getMessage().setBody(objectMapper.writeValueAsString(workflowEventDTO));
+
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: Removed

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java [hunk 2]
```diff
@@ -112,9 +137,57 @@ private void processCompleteAsProcessed(Exchange exchange) throws JsonProcessing
...
+		String message = (String) exchange.getMessage().getBody();
+		JsonObject json = new JsonObject(message);
+		WorkflowInternalActionDTO workflowEventDTO = new WorkflowInternalActionDTO();
+
+		workflowEventDTO.setResumeTimestamp(DateUtils.formatToISOString(
+				DateUtils.getUTCCurrentDateTime()
+						.plusSeconds(Long.parseLong((String) exchange.getProperty("pauseFor")))));
+		workflowEventDTO.setRid(json.getString("rid"));
+		workflowEventDTO.setDefaultResumeAction(WorkflowActionCode.STOP_PROCESSING.toString());
+		workflowEventDTO.setActionCode(WorkflowInternalActionCode.PAUSED_FOR_ADDITIONAL_INFO.toString());
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: Fixed

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java [hunk 3]
```diff
@@ -112,9 +137,57 @@ private void processCompleteAsProcessed(Exchange exchange) throws JsonProcessing
...
+	}
 
+	private void processResumeParentFlow(Exchange exchange) throws JsonProcessingException {
+		String message = (String) exchange.getMessage().getBody();
+		JsonObject json = new JsonObject(message);
+		WorkflowInternalActionDTO workflowEventDTO = new WorkflowInternalActionDTO();
+		workflowEventDTO.setRid(json.getString("rid"));
+		workflowEventDTO.setActionCode(WorkflowInternalActionCode.RESUME_PARENT_FLOW.toString());
+		workflowEventDTO.setActionMessage(PlatformSuccessMessages.PACKET_RESUME_PARENT_FLOW.getMessage());
+		workflowEventDTO.setReg_type(json.getString("reg_type"));
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: Added in JsonConstant

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java [hunk 4]
```diff
@@ -112,9 +137,57 @@ private void processCompleteAsProcessed(Exchange exchange) throws JsonProcessing
...
+
+		workflowEventDTO.setResumeTimestamp(DateUtils.formatToISOString(
+				DateUtils.getUTCCurrentDateTime()
+						.plusSeconds(Long.parseLong((String) exchange.getProperty("pauseFor")))));
+		workflowEventDTO.setRid(json.getString("rid"));
+		workflowEventDTO.setDefaultResumeAction(WorkflowActionCode.STOP_PROCESSING.toString());
+		workflowEventDTO.setActionCode(WorkflowInternalActionCode.PAUSED_FOR_ADDITIONAL_INFO.toString());
+		workflowEventDTO.setEventTimestamp(DateUtils.formatToISOString(DateUtils.getUTCCurrentDateTime()));
+		workflowEventDTO.setActionMessage(PlatformSuccessMessages.PACKET_PAUSED_FOR_ADDITIONAL_INFO.getMessage());
+		workflowEventDTO.setSubProcess((String) exchange.getProperty("subProcess"));
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 
 
+// TODO: Auto-generated Javadoc
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java]: fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/workflow/dto/WorkflowPausedForAdditionalInfoEventDTO.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	private String instanceId;
+
+	/** The result code. */
+	private String resultCode;
+
+	/** The workflow type. */
+	private String workflowType;
+
+	/** The sub process. */
+	private String subProcess;
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/workflow/dto/WorkflowPausedForAdditionalInfoEventDTO.java]: fixed

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/config/WorkflowManagerConfigBeans.java [hunk 7]
```diff
@@ -89,4 +90,9 @@ public RestTemplate getRestTemplate() {
 	public WorkflowActionJob getWorkflowActionJob() {
 		return new WorkflowActionJob();
 	}
+
+	@Bean
+	public PublisherClient<String, WorkflowPausedForAdditionalInfoEventDTO, HttpHeaders> getWebPublisherClientForPausedForAdditionalInfo() {
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/config/WorkflowManagerConfigBeans.java]: fixed

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 8]
```diff
@@ -300,4 +324,99 @@ private void sendWorkflowCompletedWebSubEvent(InternalRegistrationStatusDto regi
...
+						workflowInternalActionDTO.getReg_type(), workflowInternalActionDTO.getIteration());
+		if (subWorkflowDtos != null && !subWorkflowDtos.isEmpty()) {
+			SubWorkflowDto subWorkflowDto = subWorkflowDtos.get(0);
+			InternalRegistrationStatusDto registrationStatusDto = registrationStatusService.getRegistrationStatus(
+					subWorkflowDto.getRegId(), subWorkflowDto.getParentProcess(), subWorkflowDto.getParentIteration());
+			List<InternalRegistrationStatusDto> internalRegistrationStatusDtos = new ArrayList<InternalRegistrationStatusDto>();
+			internalRegistrationStatusDtos.add(registrationStatusDto);
+			workflowActionService.processWorkflowAction(internalRegistrationStatusDtos,
+					WorkflowActionCode.RESUME_FROM_BEGINNING.toString());
+		}
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java:345]: added

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 9]
```diff
@@ -300,4 +324,99 @@ private void sendWorkflowCompletedWebSubEvent(InternalRegistrationStatusDto regi
...
+		String additionalRequestId = workflowInternalActionDTO.getRid() + "-"
+				+ workflowInternalActionDTO.getSubProcess() + "-" + iteration;
+		return additionalRequestId;
+	}
+
+	private void sendWorkflowPausedForAdditionalInfoEvent(InternalRegistrationStatusDto registrationStatusDto,
+			String additonalInfoRequestId, String subProcess) {
+		WorkflowPausedForAdditionalInfoEventDTO workflowPausedForAdditionalInfoEventDTO = new WorkflowPausedForAdditionalInfoEventDTO();
+		workflowPausedForAdditionalInfoEventDTO.setInstanceId(registrationStatusDto.getRegistrationId());
+		workflowPausedForAdditionalInfoEventDTO.setResultCode(registrationStatusDto.getStatusCode());
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: removed as it is not needed

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: why ADDITIONAL_INFO_PROCESS is not in Json Constants but PAUSE_FOR is there?

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java]: I can still see the TODO comment

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java [hunk 10]
```diff
@@ -43,8 +38,8 @@ public WorkflowInternalActionDTO() {
 	/** The resume remove tags. */
 	private String resumeRemoveTags;
 
-
-
+	/** The sub process. */
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java]: Change the comment also to additional info process

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java [hunk 11]
```diff
@@ -1,14 +1,33 @@
 package io.mosip.registration.processor.core.code;
 
+// TODO: Auto-generated Javadoc
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java]: remove this TODO

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java]: fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java (hunk 10 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/abstractverticle/WorkflowInternalActionDTO.java]: fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java (hunk 11 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java]: fixed

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**